test/github/kv_test.rb

```diff
@@ -54,6 +54,159 @@ def test_set_failure
+
+  def test_increment_large_value
+    result = @kv.increment("foo", amount: 10000)
+
+    assert_equal 10000, result
+  end
+
+  def test_increment_negative
+    result = @kv.increment("foo", amount: -1)
+
+    assert_equal -1, result
+  end
+
+  def test_increment_negative_to_0
+    @kv.set("foo", "1")
+    result = @kv.increment("foo", amount: -1)
+
+    assert_equal 0, result
+  end
+
+  def test_increment_multiple
+    @kv.increment("foo")
+    result = @kv.increment("foo")
+
+    assert_equal 2, result
+  end
+
+  def test_increment_multiple_different_values
+    @kv.increment("foo")
+    result = @kv.increment("foo", amount: 2)
+
+    assert_equal 3, result
+  end
+
+  def test_increment_existing
+    @kv.set("foo", "1")
+
+    result = @kv.increment("foo")
+
+    assert_equal 2, result
+  end
+
+  def test_increment_overwrites_expired_value
+    @kv.set("foo", "100", expires: 1.hour.ago)
+    expires = 1.hour.from_now.utc
+    result = @kv.increment("foo", expires: expires)
+
+    assert_equal expires.to_i, @kv.ttl("foo").value!.to_i
+    assert_equal 1, result
+  end
+
+  def test_increment_overwrites_expired_value_from_incrementing
+    @kv.increment("foo", expires: 1.hour.ago)
+    expires = 1.hour.from_now.utc
+    result = @kv.increment("foo", expires: expires)
+
+    assert_equal 1, result
+    assert_equal expires.to_i, @kv.ttl("foo").value!.to_i
+  end
+
+  def test_increment_overwrites_expired_with_nil
+    @kv.increment("foo", expires: 1.hour.ago)
+    result = @kv.increment("foo")
+
+    assert_equal 1, result
+    assert_nil @kv.ttl("foo").value!
+  end
+
+  def test_increment_sets_expires
+    expires = 1.hour.from_now.utc
+    @kv.increment("foo", expires: expires)
+
+    assert_equal expires.to_i, @kv.ttl("foo").value!.to_i
+  end
+
+  def test_increment_sets_expires_only_on_insert
+    expires = 1.hour.from_now.utc
+    result = @kv.increment("foo", expires: expires, touch_on_insert: true)
+    assert_equal 1, result
+
+    result = @kv.increment("foo", expires: 3.hours.from_now.utc, touch_on_insert: true)
+    assert_equal 2, result
+
+    assert_equal expires.to_i, @kv.ttl("foo").value!.to_i
+  end
+
+  def test_increment_sets_expires_only_on_insert_for_existing
+    expires = 1.hour.from_now.utc
+    @kv.set("foo", "100", expires: expires)
+
+    result = @kv.increment("foo", expires: 3.hours.from_now.utc, touch_on_insert: true)
+    assert_equal 101, result
+
+    assert_equal expires.to_i, @kv.ttl("foo").value!.to_i
+  end
+
+  def test_increment_updates_expires
+    expires = 2.hours.from_now.utc
+
+    @kv.set("foo", "100", expires: 1.hour.from_now)
+    result = @kv.increment("foo", expires: expires)
+
+    assert_equal 101, result
+    assert_equal expires.to_i, @kv.ttl("foo").value!.to_i
+  end
+
+  def test_increment_non_integer_key_value
+    @kv.set("foo", "bar")
+
+    assert_raises GitHub::KV::InvalidValueError do
+      @kv.increment("foo")
+    end
+    assert_equal "bar", @kv.get("foo").value!
+  end
```

Comment:
Is it possible to reinitialize the client in tests so we can explicitly test with and without the `Mysql2::Client::FOUND_ROWS` flag? If not, can we setup another CI build with the opposite configuration? That might be preferable, so we can get all the tests in both states.

Reply:
Since the tests use active record I'm not sure we can easily change this: https://github.com/rails/rails/pull/25370

We might be able to get away with monkey patching active records before the connection is initialized but I'm not completely sure how to do that.

Reply:
Monkey patching AR in tests might be okay. Maybe we can add an environment variable like `GITHUB_DS_OVERRIDE_MYSQL_FOUND_ROWS` and do the monkey patch if that's set. We can add something to the travis build matrix to run the whole test suite with and without this variable set.

Reply:
I can help later with the monkey patching if you want. It might be a bit nasty.

Reply:
Would love to 🍐 on that 👍

Reply:
Not sure how hard this would be but if we're going to be modifying the CI what about moving it over to Actions while we're at it? Would be neat to play with that a bit - I don't think it would be too difficult.

Reply:
> The desirable configuration is to not have the flag set. Unfortunately GitHub does have it set (for historical reasons)

So, you're saying that GitHub has the flag set and that's the default for ActiveRecord and it's hard for people to disable manually? If that's the case, let's just raise an error if `#found_rows_is_set?` is `false`. This library is only meant to work with MySQL and ActiveRecord, so I don't think it would be problematic to raise an error if someone's gone to great lengths to unset this flag.